jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -828,6 +849,154 @@ MethodHandle insertArguments(MethodHandle equivalent, MethodHandle unboxingHandl
 	private static final int BSM_TYPE_ARGUMENT_INDEX = 2;
 	private static final int BSM_OPTIONAL_ARGUMENTS_START_INDEX = 3;
 
+	@SuppressWarnings("unused")
+	private static final Object resolveConstantDynamic(long j9class, String name, String fieldDescriptor, long bsmData) throws Throwable {
+		Object result = null;
+		Class<?> typeClass = null;
+
+		try {
+			VMLangAccess access = VM.getVMLangAccess();
+			Object internalRamClass = access.createInternalRamClass(j9class);
+			Class<?> classObject = null;
+			if (JITHELPERS.is32Bit()) {
+				classObject = JITHELPERS.getClassFromJ9Class32((int)j9class);
+			} else {
+				classObject = JITHELPERS.getClassFromJ9Class64(j9class);
+			}
+
+			Objects.requireNonNull(classObject);
```

Comment:
Please add a message to this throwable to indicate the j9class, name, etc so that issues can more easily be debugged

Reply:
added ExternalMessage in the catch block